### PR TITLE
Kevin Marker

### PR DESCRIPTION
**Pull request by kmarker1101**

### CODE REVIEW

File: assignments/prototypes.js [hunk 1]
```diff
@@ -102,9 +135,80 @@
...
+
+Villian.prototype = Object.create(Humanoid.prototype);
+
+Villian.prototype.maulsAttack = function () {
+    aragorn.hp -= 5;
+    if (witchKing.hp <= 0) {
+        return aragorn.destroy();
+    }
+    return `Remove 5 HP from hero. The life of ${aragorn.name} is ${aragorn.hp}`;
+};
```
Comment (johnoro) [inline assignments/prototypes.js]: It's be better to use `this` and pass in the attacked as an object in order to make this reusable:
```javascript
Villian.prototype.maulsAttack = function (attacked) {
    attacked.hp -= 5;
    if (this.hp <= 0) {
        return attacked.destroy();
    }
    return `Remove 5 HP from hero. The life of ${attacked.name} is ${attacked.hp}`;
};
```

Comment (johnoro) [reply assignments/prototypes.js]: The same applies to the `Hero`'s class attack method.

**Review by johnoro — COMMENTED** [review]
Great job!

File: assignments/prototypes.js (hunk 1 above)
Comment (kmarker1101) [reply assignments/prototypes.js]: I abstracted to use attacked but was still unable to use `this`. It breaks the if statement.

**Pull request closed without merging**